Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -63,16 +63,20 @@ def run
 
         exec_editor(*paths)
 
+        is_formula = T.let(false, T::Boolean)
         if paths.any? do |path|
              next if path == "--project"
 
-             !Homebrew::EnvConfig.no_install_from_api? &&
-             !Homebrew::EnvConfig.no_env_hints? &&
-             (core_formula_path?(path) || core_cask_path?(path) || core_formula_tap?(path) || core_cask_tap?(path))
+             break if Homebrew::EnvConfig.no_install_from_api?
+             break if Homebrew::EnvConfig.no_env_hints?
```

Comment:
Maybe move it into `if` above? Also not sure that `no_install_from_api` is still relevant.

Reply:
> Maybe move it into `if` above?

Good idea, works for me.

> Also not sure that `no_install_from_api` is still relevant.

If it's set: no need to tell people to do `HOMEBREW_NO_INSTALL_FROM_API=1`.

Reply:
@abitrolly can you take a look at this? thanks!